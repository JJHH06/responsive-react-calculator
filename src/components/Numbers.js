import React, {useEffect, useState} from 'react'

const Numbers = ({numero, calculateNumbers}) =>{
    const clickNumberListener = e =>{
        calculateNumbers(numero);
    }
    return (
        <div className = "col calculator-numbers" onClick  = {clickNumberListener}>
            <p className = "calculator-appenders">{numero}</p>
        </div>
    )
}

export default Numbers