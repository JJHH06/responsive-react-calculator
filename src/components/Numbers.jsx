/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
// se desactivaron las reglas que no dejan que funcionen los botones
import React from 'react';
import PropTypes from 'prop-types';

const Numbers = ({ numero, calculateNumbers }) => {
  const clickNumberListener = () => {
    calculateNumbers(numero);
  };
  return (
    <div className="col calculator-numbers" onClick={clickNumberListener}>
      <p className="calculator-appenders">{numero}</p>
    </div>
  );
};

Numbers.propTypes = {
  numero: PropTypes.string.isRequired,
  calculateNumbers: PropTypes.func.isRequired,
};

export default Numbers;
