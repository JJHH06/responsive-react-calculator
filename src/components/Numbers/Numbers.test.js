import React from 'react';
import ReactDOM from 'react-dom';
import { isTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Numbers from './Numbers';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Numbers />, div);
});

it('renders Number correctly', () => {
  const { getByTestId } = render(<Numbers numero="0" />);
  expect(getByTestId('number')).toHaveTextContent('0');
});

it('renders second Number correctly', () => {
  const { getByTestId } = render(<Numbers numero="2" />);
  expect(getByTestId('number')).toHaveTextContent('2');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Numbers numero="2" />).toJSON();
  expect(tree).toMatchSnapshot();
});
