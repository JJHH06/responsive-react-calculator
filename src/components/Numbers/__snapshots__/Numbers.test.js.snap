// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col calculator-numbers"
  data-testid="number"
  onClick={[Function]}
>
  <p
    className="calculator-appenders"
  >
    2
  </p>
</div>
`;
